File: benches/knight.rs
#![feature(test)]

extern crate test;
extern crate hexchess;

use hexchess::hexchess::hexchess::Hexchess;
use hexchess::constants::Color;
use hexchess::hexchess::pieces::knight::knight_moves_unsafe;
use test::Bencher;

#[bench]
fn bench_knight_moves_unsafe(b: &mut Bencher) {
    let hexchess = Hexchess::init();
    b.iter(|| {
        for n in 0..90 {
            knight_moves_unsafe(&hexchess, n, &Color::White);
        }
    });
}

#[bench]
fn bench_knight_moves_unsafe_black(b: &mut Bencher) {
    let hexchess = Hexchess::init();
    b.iter(|| {
        // Test moves from different positions to get a good average
        for n in 0..90 {
            knight_moves_unsafe(&hexchess, n, &Color::White);
        }
    });
} 

File: php/src/Pieces/Knight.php
<?php

namespace Bedard\Hexchess\Pieces;

use Bedard\Hexchess\Board;
use Bedard\Hexchess\Constants;
use Bedard\Hexchess\Hexchess;
use Bedard\Hexchess\San;

class Knight
{
    public static function moves(Hexchess $hexchess, int $from, string $color): array
    {
        $moves = [];

        $graph = Constants::KNIGHT_GRAPH[$from];

        foreach ($graph as $to) {
            if ($hexchess->board[$to] === null || Board::color($hexchess->board[$to]) !== $color) {
                $moves[] = new San($from, $to);
            }
        }

        return $moves;
    }
}
